crates/sui-graphql-client/src/lib.rs

```diff
@@ -393,14 +405,14 @@ impl Client {
                     None,
                 ).await?;
 
-                if let Some(page) = response {
-                    for object in page.data {
-                        if let Some(coin) = Coin::try_from_object(&object) {
+                if !response.is_empty() {
```

Comment:
the point of the simplification is that you no longer need this outer check. This code can be written as:

```rust
for object in response.data() {
    if let Some(coin) = Coin::try_from_object(object) {
      yield coin.into_owned();
    }
}

after = response.page_info.end_cursor;
if after.is_none() {
  break;
}
```